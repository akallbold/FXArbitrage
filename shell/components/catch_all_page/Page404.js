import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'redux-little-router'
import { Button } from 'playa-styleguide'
import Robot404 from './Robot404'
import Styles from './ErrorPage.css'

const Page404 = ({ prevPathname }) => (
  <div className={Styles.errorPage}>
    <Robot404 className={Styles.image} />
    <div className={Styles.message}>This is not the page youâ€™re looking for...</div>
    {prevPathname &&
      <Button
        className={Styles.button}
        href={prevPathname}
        component={Link}
        priority='primary'
        size='large'
      >
        Take me back
      </Button>
    }
    {!prevPathname &&
      <div className={Styles.message}>
        Please use the brands menu at the top of the page to continue.
      </div>
    }
  </div>
)

Page404.propTypes = {
  prevPathname: PropTypes.string
}

Page404.defaultProps = {
  prevPathname: null
}

export default Page404
