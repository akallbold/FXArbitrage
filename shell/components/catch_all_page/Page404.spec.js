import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Page404 from './Page404'

describe('<Page404 />', () => {
  it('should render properly given all props', () => {
    const component = shallow(
      <Page404 prevPathname='some/path/here' />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render a button with prevPathname as href', () => {
    const component = shallow(
      <Page404 prevPathname='some/path/here' />
    )
    expect(component.find('Button').prop('href')).toEqual('some/path/here')
  })

  it('should render a message instead of button with undefined prevPathname', () => {
    const component = shallow(<Page404 />)

    expect(component.find('Button').length).toEqual(0)
    expect(component.find('.message').length).toEqual(2)
  })
})
