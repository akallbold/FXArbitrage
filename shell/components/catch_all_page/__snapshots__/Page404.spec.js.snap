// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page404 /> should render properly given all props 1`] = `
<div
  className="errorPage"
>
  <Robot404
    className="image"
  />
  <div
    className="message"
  >
    This is not the page youâ€™re looking for...
  </div>
  <Button
    className="button"
    component={[Function]}
    disabled={false}
    href="some/path/here"
    onClick={null}
    priority="primary"
    size="large"
  >
    Take me back
  </Button>
</div>
`;
